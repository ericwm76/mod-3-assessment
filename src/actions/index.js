export const setOrders = orders => ({
  type: 'SET_ORDERS',
  orders
});

export const addOrder = (name, ingredients) => ({
  type: 'ADD_ORDER',
  order: {
    name, ingredients
  }
})