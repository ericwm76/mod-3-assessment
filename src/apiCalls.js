export const getOrders = () => {
  return fetch('http://localhost:3001/api/v1/orders')
      .then(response => response.json())
}

export const postOrder = async (name, ingredients) => {
  const options = {
    method: 'POST',
    body: JSON.stringify({
      name,
      ingredients
    }),
    headers: {
      'Content-Type': 'application/json'  
    }
  }

  let response = await fetch('http://localhost:3001/api/v1/orders', options);

  if(!response.ok) {
    console.log(response.status)
  }

  let data = response.json();

  return data;
}