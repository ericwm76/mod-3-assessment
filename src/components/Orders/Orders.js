import React from 'react';
import './Orders.css';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setOrders } from '../../actions';
import { getOrders } from '../../apiCalls';

class Orders extends React.Component {
  constructor(props) {
    super();
    this.state = {

    }
  }

  displayOrders = () => {
    const orderEls = props.orders.map(order => {
      return (
        <div className="order">
          <h3>{order.name}</h3>
          <ul className="ingredient-list">
            {order.ingredients.map(ingredient => {
              return <li>{ingredient}</li>
            })}
          </ul>
        </div>
      )
    });
  
  return (
    <section>
      { orderEls.length ? orderEls : <p>No orders yet!</p> }
    </section>
  )
}
}
}

export const mapStateToProps = ({ orders }) => ({
  orders
});

export const mapDispatchToProps = dispatch => (
  bindActionCreators({
    setOrders,
  }, dispatch)
);

export default connect(mapStateToProps, mapDispatchToProps)(Orders);