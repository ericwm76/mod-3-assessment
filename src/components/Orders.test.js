import React from 'react';
import { Orders, mapStateToProps, mapDispatchToProps } from './Orders/Orders';
import { shallow } from 'enzyme';
import { getOrders } from '../apiCalls';
import { setOrders } from '../actions';

jest.mock('../apiCalls');
jest.mock('../actions');

describe('Orders', () => {
  let wrapper, mockOrders;

  beforeEach(() => {

    mockOrders = [
      { id: 1, name: 'Carlos', ingredients: ['lettuce', 'queso', 'steak']   },
      { id: 2, name: 'Regina', ingredients: ['beans', 'queso', 'cilantro']   },
      { id: 3, name: 'Sofia', ingredients: ['rice', 'sour cream', 'carnitas']   }
    ];

    wrapper = shallow(<Orders setOrders={jest.fn()} orders={mockOrders}/>);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call getOrders on mount', () => {
    expect(getOrders).toHaveBeenCalled();
  })

  it('should call setOrders on mount', () => {
    expect(setOrders).toHaveBeenCalled();
  })

  describe('mapStateToProps', () => {
    it('should return an object with the orders array', () => {
      const mockState = {
        orders: mockOrders,
        filter: 'SET_PLAYERS'
      };
      const expected = {
        orders: mockOrders
      };

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected);
    })
  })

  describe('mapDispatchToProps', () => {
    it('calls dispatch with a setOrders action', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = setOrders(mockOrders);

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.setOrders(mockOrders);

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})