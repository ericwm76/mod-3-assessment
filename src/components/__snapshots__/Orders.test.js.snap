// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Orders should match the snapshot 1`] = `
<section>
  <div
    className="order"
    key="1"
  >
    <h3>
      Carlos
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        lettuce
      </li>
      <li
        key="1"
      >
        queso
      </li>
      <li
        key="2"
      >
        steak
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="2"
  >
    <h3>
      Regina
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        beans
      </li>
      <li
        key="1"
      >
        queso
      </li>
      <li
        key="2"
      >
        cilantro
      </li>
    </ul>
  </div>
  <div
    className="order"
    key="3"
  >
    <h3>
      Sofia
    </h3>
    <ul
      className="ingredient-list"
    >
      <li
        key="0"
      >
        rice
      </li>
      <li
        key="1"
      >
        sour cream
      </li>
      <li
        key="2"
      >
        carnitas
      </li>
    </ul>
  </div>
</section>
`;
